### PR TITLE
Release1.1

### PR DESCRIPTION
**Pull request by lnagle**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "ddq-mysql",
-    "version": "1.0.0",
+    "version": "1.0.1",
```
Comment (GoogilyBoogily) [inline package.json]: These changes aren't backwards compatible. So the version bump isn't correct.

File: lib/ddq-backend-mysql.js [hunk 2]
```diff
@@ -398,3 +391,4 @@ module.exports = (configValidation, crypto, EventEmitter, mysql, timers) => {
     return DdqBackendMySql;
 };
 
+
```
Comment (fidian) [inline lib/ddq-backend-mysql.js]: Was this accidental?

File: lib/ddq-backend-mysql.js [hunk 3]
```diff
@@ -331,8 +331,10 @@ module.exports = (configValidation, crypto, EventEmitter, mysql, timers) => {
         /**
          * Resets the relevant flags and clears the timers, which stops both
          * polling and restoring.
+         *
+         * @param {Function} [callback]
          */
-        pausePolling() {
+        stopListening(callback) {
```
Comment (fidian) [inline lib/ddq-backend-mysql.js:337]: This function, due to a race condition, can complete and call the callback.  After successfully `stopListening` the ddq core could get an event with a wrapped message.

To close this hole, you'll have to wrap the emit in a condition checking if you are still polling.  But I don't like that solution.  I would rather have stopListening stall until we know it is certainly stopped, especially if we intend to close the connection as well.  ... Thinking.

**Review by fidian — CHANGES_REQUESTED** [review]

File: lib/ddq-backend-mysql.js (hunk 2 above)
Comment (lnagle) [reply lib/ddq-backend-mysql.js]: Yes; removed.

File: package.json (hunk 1 above)
Comment (lnagle) [reply package.json]: Fixed.

**Comment by AbsentSemicolon** [comment]
Looks good to me though we need to figure out why the tests are failing in Travis.

**Comment by GoogilyBoogily** [comment]
Looks solid.

File: lib/ddq-backend-mysql.js (hunk 3 above)
Comment (fidian) [reply lib/ddq-backend-mysql.js:337]: We talked a bit about this.  I think I'll discuss it more with you, but the essence of the solution is to, I believe, track a few more things in DDQ and pull more logic out of the plugins.  Let DDQ coordinate the callbacks for starting/stopping/etc.  The plugin shouldn't even worry about `this.currentlyRestoring` nor `this.currentlyPolling` in the way it's currently approaching that problem.  It should track when it is doing an operation, but probably not when it is stalling.

**Review by fidian — CHANGES_REQUESTED** [review]

**Pull request closed without merging**